File: org.summer.view.widget/src/org/summer/view/widget/IInputElement.java
package org.summer.view.widget;
public interface IInputElement
	{
		void AddHandler (RoutedEvent routedEvent, Delegate handler);
		boolean CaptureMouse ();
		boolean CaptureStylus ();
		boolean Focus ();
		void RaiseEvent (RoutedEventArgs e);
		void ReleaseMouseCapture ();
		void ReleaseStylusCapture ();
		void RemoveHandler (RoutedEvent routedEvent, Delegate handler);

		boolean Focusable { get; set; }
		boolean IsEnabled { get; }
		boolean IsKeyboardFocused { get; }
		boolean IsKeyboardFocusWithin { get; }
		boolean IsMouseCaptured { get; }
		boolean IsMouseDirectlyOver { get; }
		boolean IsMouseOver { get; }
		boolean IsStylusCaptured { get; }
		boolean IsStylusDirectlyOver { get; }
		boolean IsStylusOver { get; }

		/*event*/ KeyboardFocusChangedEventHandler GotKeyboardFocus;
		/*event*/ MouseEventHandler GotMouseCapture;
		/*event*/ StylusEventHandler GotStylusCapture;
		/*event*/ KeyEventHandler KeyDown;
		/*event*/ KeyEventHandler KeyUp;
		/*event*/ KeyboardFocusChangedEventHandler LostKeyboardFocus;
		/*event*/ MouseEventHandler LostMouseCapture;
		/*event*/ StylusEventHandler LostStylusCapture;
		/*event*/ MouseEventHandler MouseEnter;
		/*event*/ MouseEventHandler MouseLeave;
		/*event*/ MouseButtonEventHandler MouseLeftButtonDown;
		/*event*/ MouseButtonEventHandler MouseLeftButtonUp;
		/*event*/ MouseEventHandler MouseMove;
		/*event*/ MouseButtonEventHandler MouseRightButtonDown;
		/*event*/ MouseButtonEventHandler MouseRightButtonUp;
 		/*event*/ MouseWheelEventHandler MouseWheel;
		/*event*/ KeyboardFocusChangedEventHandler PreviewGotKeyboardFocus;
		/*event*/ KeyEventHandler PreviewKeyDown;
		/*event*/ KeyEventHandler PreviewKeyUp;
		/*event*/ KeyboardFocusChangedEventHandler PreviewLostKeyboardFocus;
		/*event*/ MouseButtonEventHandler PreviewMouseLeftButtonDown;
		/*event*/ MouseButtonEventHandler PreviewMouseLeftButtonUp;
		/*event*/ MouseEventHandler PreviewMouseMove;
		/*event*/ MouseButtonEventHandler PreviewMouseRightButtonDown;
		/*event*/ MouseButtonEventHandler PreviewMouseRightButtonUp;
		/*event*/ MouseWheelEventHandler PreviewMouseWheel;
		/*event*/ StylusButtonEventHandler PreviewStylusButtonDown;
		/*event*/ StylusButtonEventHandler PreviewStylusButtonUp;
		/*event*/ StylusDownEventHandler PreviewStylusDown;
		/*event*/ StylusEventHandler PreviewStylusInAirMove;
		/*event*/ StylusEventHandler PreviewStylusInRange;
		/*event*/ StylusEventHandler PreviewStylusMove;
		/*event*/ StylusEventHandler PreviewStylusOutOfRange;
		/*event*/ StylusSystemGestureEventHandler PreviewStylusSystemGesture;
		/*event*/ StylusEventHandler PreviewStylusUp;
		/*event*/ TextCompositionEventHandler PreviewTextInput;
		/*event*/ StylusButtonEventHandler StylusButtonDown;
		/*event*/ StylusButtonEventHandler StylusButtonUp;
		/*event*/ StylusDownEventHandler StylusDown;
		/*event*/ StylusEventHandler StylusEnter;
		/*event*/ StylusEventHandler StylusInAirMove;
		/*event*/ StylusEventHandler StylusInRange;
		/*event*/ StylusEventHandler StylusLeave;
		/*event*/ StylusEventHandler StylusMove;
		/*event*/ StylusEventHandler StylusOutOfRange;
		/*event*/ StylusSystemGestureEventHandler StylusSystemGesture;
		/*event*/ StylusEventHandler StylusUp;
		/*event*/ TextCompositionEventHandler TextInput;
	}

File: summer/WebContent/js/system/windows/IInputElement.js
/**
 * IInputElement
 */

define(["dojo/_base/declare", "system/Type"], 
		function(declare, Type){
	var IInputElement = declare("IInputElement", Object,{
		constructor:function(){
		}
	});
	
	
	IInputElement.Type = new Type("IInputElement", IInputElement, [Object.Type], true);
	return IInputElement;
});
//	public interface IInputElement
//	{
//		event MouseButtonEventHandler PreviewMouseLeftButtonDown;
//		event MouseButtonEventHandler MouseLeftButtonDown;
//		event MouseButtonEventHandler PreviewMouseLeftButtonUp;
//		event MouseButtonEventHandler MouseLeftButtonUp;
//		event MouseButtonEventHandler PreviewMouseRightButtonDown;
//		event MouseButtonEventHandler MouseRightButtonDown;
//		event MouseButtonEventHandler PreviewMouseRightButtonUp;
//		event MouseButtonEventHandler MouseRightButtonUp;
//		event MouseEventHandler PreviewMouseMove;
//		event MouseEventHandler MouseMove;
//		event MouseWheelEventHandler PreviewMouseWheel;
//		event MouseWheelEventHandler MouseWheel;
//		event MouseEventHandler MouseEnter;
//		event MouseEventHandler MouseLeave;
//		event MouseEventHandler GotMouseCapture;
//		event MouseEventHandler LostMouseCapture;
//		event StylusDownEventHandler PreviewStylusDown;
//		event StylusDownEventHandler StylusDown;
//		event StylusEventHandler PreviewStylusUp;
//		event StylusEventHandler StylusUp;
//		event StylusEventHandler PreviewStylusMove;
//		event StylusEventHandler StylusMove;
//		event StylusEventHandler PreviewStylusInAirMove;
//		event StylusEventHandler StylusInAirMove;
//		event StylusEventHandler StylusEnter;
//		event StylusEventHandler StylusLeave;
//		event StylusEventHandler PreviewStylusInRange;
//		event StylusEventHandler StylusInRange;
//		event StylusEventHandler PreviewStylusOutOfRange;
//		event StylusEventHandler StylusOutOfRange;
//		event StylusSystemGestureEventHandler PreviewStylusSystemGesture;
//		event StylusSystemGestureEventHandler StylusSystemGesture;
//		event StylusButtonEventHandler StylusButtonDown;
//		event StylusButtonEventHandler PreviewStylusButtonDown;
//		event StylusButtonEventHandler PreviewStylusButtonUp;
//		event StylusButtonEventHandler StylusButtonUp;
//		event StylusEventHandler GotStylusCapture;
//		event StylusEventHandler LostStylusCapture;
//		event KeyEventHandler PreviewKeyDown;
//		event KeyEventHandler KeyDown;
//		event KeyEventHandler PreviewKeyUp;
//		event KeyEventHandler KeyUp;
//		event KeyboardFocusChangedEventHandler PreviewGotKeyboardFocus;
//		event KeyboardFocusChangedEventHandler GotKeyboardFocus;
//		event KeyboardFocusChangedEventHandler PreviewLostKeyboardFocus;
//		event KeyboardFocusChangedEventHandler LostKeyboardFocus;
//		event TextCompositionEventHandler PreviewTextInput;
//		event TextCompositionEventHandler TextInput;
//		bool IsMouseOver
//		{
//			get;
//		}
//		bool IsMouseDirectlyOver
//		{
//			get;
//		}
//		bool IsMouseCaptured
//		{
//			get;
//		}
//		bool IsStylusOver
//		{
//			get;
//		}
//		bool IsStylusDirectlyOver
//		{
//			get;
//		}
//		bool IsStylusCaptured
//		{
//			get;
//		}
//		bool IsKeyboardFocusWithin
//		{
//			get;
//		}
//		bool IsKeyboardFocused
//		{
//			get;
//		}
//		bool IsEnabled
//		{
//			get;
//		}
//		bool Focusable
//		{
//			get;
//			set;
//		}
//		void RaiseEvent(RoutedEventArgs e);
//		void AddHandler(RoutedEvent routedEvent, Delegate handler);
//		void RemoveHandler(RoutedEvent routedEvent, Delegate handler);
//		bool CaptureMouse();
//		void ReleaseMouseCapture();
//		bool CaptureStylus();
//		void ReleaseStylusCapture();
//		bool Focus();
//	}